### PR TITLE
Replace use of audio_format objects

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Comment by ranj063** [comment]
SOFCI TEST

File: tools/topology/topology2/platform/intel/sdw-jack-generic.conf [hunk 1]
```diff
@@ -186,56 +186,103 @@ Object.Pipeline.host-gateway-capture [
...
+			CombineArrays.Object.Base.output_audio_format [
+				# array of 2ch formats with range of out valid bit depths
+				{
+					out_bit_depth [
+						32
+					]
+					out_valid_bit_depth [
+						24
+						32
+					]
```
Comment (plbossart) [inline tools/topology/topology2/platform/intel/sdw-jack-generic.conf:224]: is 

````
out_channels [					
    2
]
````

missing or default?

File: tools/topology/topology2/platform/intel/sdw-jack-generic.conf [hunk 2]
```diff
@@ -186,56 +186,103 @@ Object.Pipeline.host-gateway-capture [
...
+					out_ch_cfg [
+						$CHANNEL_CONFIG_MONO
+					]
+					out_ch_map [
+						$CHANNEL_MAP_MONO
+					]
+				}
+				# 16-bit 2ch
+				{
+					out_bit_depth		16
```
Comment (plbossart) [inline tools/topology/topology2/platform/intel/sdw-jack-generic.conf:266]: is 
````
out_bit_depth		16
````
the same as 
````
out_bit_depth	[
	16
]
````
if yes, why not always use the former case for single-value configurations?

File: tools/topology/topology2/platform/intel/sdw-jack-generic.conf [hunk 3]
```diff
@@ -186,56 +186,103 @@ Object.Pipeline.host-gateway-capture [
 		Object.Widget.host-copier.1 {
 			stream_name	"Passthrough Capture 0"
 			pcm_id 1
-			num_input_audio_formats 9
+			num_input_audio_formats 3
```
Comment (plbossart) [inline tools/topology/topology2/platform/intel/sdw-jack-generic.conf:189]: typo in commit message: introduces

File: tools/topology/topology2/platform/intel/sdw-jack-generic.conf (hunk 1 above)
Comment (ranj063) [reply tools/topology/topology2/platform/intel/sdw-jack-generic.conf:222]: default

File: tools/topology/topology2/platform/intel/sdw-jack-generic.conf (hunk 2 above)
Comment (ranj063) [reply tools/topology/topology2/platform/intel/sdw-jack-generic.conf]: this is a feature or a limitation with the way the CombineArrays feature has been defined in the compilet today. So all values even single values have to be specified with an array. I've updated the patch to use array consistenly now

**Review by plbossart — APPROVED** [review]

**Comment by fredoh9** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
> looks good but I wonder if we should rootcause #8320 first before merging.

@kv2019i fix posted in #8323. We can merge this after that one

**Pull request merged**